scripts/apps/contacts/components/Form/ContactNumberInput.jsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Row, LineInput, Input, SelectInput, Toggle, Label} from './';
+import {MSG_REQUIRED, KEYCODES} from '../../../contacts/constants';
+import {set, get, isEmpty} from 'lodash';
+
+export class ContactNumberInput extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            preventSwitch: false,
+            touched: {},
+        };
+        this.onChange = this.onChange.bind(this);
+        this.onBlur = this.onBlur.bind(this);
+        this.isFieldInValid = this.isFieldInValid.bind(this);
+    }
+
+    onBlur(e) {
+        let _touched = this.state.touched;
+
+        set(_touched, e.target.name, true);
+
+        this.setState({touched: _touched});
+    }
+
+    isFieldInValid(field, value) {
+        return get(this.state.touched, field, false) && isEmpty(value);
+    }
+
+    onChange(field, value) {
+        // Turn off and prevent public switch for confidential contact number
+        if (value === 'Confidential') {
+            this.props.value['public'] = false;
+            this.setState({preventSwitch: true});
+        } else {
+            this.setState({preventSwitch: false});
+        }
+        this.props.onChange(field, value);
+    }
+
+    render() {
+        const {value, field, remove, onChange, readOnly, usages} = this.props;
+
+        return (
+            <Row flex={true}>
+                <LineInput readOnly={readOnly}
```

Comment:
imo we should introduce components for LineInputText, LineInputSelect etc. so it would handle the whole sd-line markup and generate the label which we can connect to the field